core/src/oauth/providers/zendesk.rs

```diff
@@ -0,0 +1,90 @@
+use crate::oauth::{
+    connection::{
+        Connection, ConnectionProvider, FinalizeResult, Provider, ProviderError, RefreshResult,
+    },
+    providers::utils::execute_request,
+};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use lazy_static::lazy_static;
+use serde_json::json;
+use std::env;
+
+lazy_static! {
+    static ref OAUTH_ZENDESK_CLIENT_ID: String = env::var("OAUTH_ZENDESK_CLIENT_ID").unwrap();
+    static ref OAUTH_ZENDESK_CLIENT_SECRET: String =
+        env::var("OAUTH_ZENDESK_CLIENT_SECRET").unwrap();
+}
+
+pub struct ZendeskConnectionProvider {}
+
+impl ZendeskConnectionProvider {
+    pub fn new() -> Self {
+        ZendeskConnectionProvider {}
+    }
+}
+
+#[async_trait]
+impl Provider for ZendeskConnectionProvider {
+    fn id(&self) -> ConnectionProvider {
+        ConnectionProvider::Zendesk
+    }
+
+    async fn finalize(
+        &self,
+        _connection: &Connection,
+        code: &str,
+        redirect_uri: &str,
+    ) -> Result<FinalizeResult, ProviderError> {
+        let body = json!({
+            "grant_type": "authorization_code",
+            "code": code,
+            "client_id": *OAUTH_ZENDESK_CLIENT_ID,
+            "client_secret": *OAUTH_ZENDESK_CLIENT_SECRET,
+            "redirect_uri": redirect_uri,
+            "scope": "tickets:read hc:read"
+        });
+
+        let req = reqwest::Client::new()
+            .post("https://d3v-dust.zendesk.com/oauth/tokens")
```

Comment:
same comment as on the other PR is that the final production ready environment?

Reply:
Just reopened the PRs yesterday without adding any reviewers for now :) Will clean that (there's a thing with their "Global oauth" concept and subdomains that I need to dig in a bit more). 
Will put those PRs as draft for now

Reply:
Yes, confirming. This is the prod endpoint. 
```
The following Global OAuth request has been received:

Organization:
Dust
Subdomain:
d3v-dust
```